torchsnapshot/storage_plugins/gcs.py

```diff
@@ -6,26 +6,46 @@
 # LICENSE file in the root directory of this source tree.
 
 import asyncio
+import functools
+import io
+import logging
 import os
-from concurrent.futures import ThreadPoolExecutor
-from functools import partial
+import random
+import time
+from typing import Awaitable, Callable, Optional, TypeVar
+from urllib.parse import quote
+
+import aiohttp
+import google.auth.exceptions
+from google._async_resumable_media.requests.download import ChunkedDownload
+from google._async_resumable_media.requests.upload import ResumableUpload
+from google.auth._default_async import default_async
+from google.auth.transport._aiohttp_requests import AuthorizedSession
 
 from torchsnapshot.io_types import IOReq, StoragePlugin
 
+logger = logging.getLogger(__name__)
 
-_DEFAULT_CONCURRENCY = 4
```

Comment:
noob q: is there somewhere we're controlling the number of files sent at a given time?